aws/src/main/java/org/apache/iceberg/aws/lakeformation/LakeFormationTemporaryCredentials.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.aws.lakeformation;
+
+import java.time.Instant;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+
+public class LakeFormationTemporaryCredentials {
```

Comment:
let's examine every class to see (1) what class could be package protected instead of `public`, (2) what class needs to implement `Serializable`. We should do a serialization check to make sure a `AwsClientFactory` with this credential is still serializable.

Reply:
Done. Added unit tests for LakeFormationAwsClientFactory serialization check.